pkg/controller/common/helpers.go

```diff
@@ -325,26 +326,56 @@ func convertIgnition32to31(ign3config ign3types.Config) (ign3_1types.Config, err
 func ValidateIgnition(ignconfig interface{}) error {
 	switch cfg := ignconfig.(type) {
 	case ign2types.Config:
-		if reflect.DeepEqual(ign2types.Config{}, cfg) {
-			return nil
-		}
-		if report := validate2.ValidateWithoutSource(reflect.ValueOf(cfg)); report.IsFatal() {
-			return errors.Errorf("invalid ignition V2 config found: %v", report)
-		}
-		return nil
+		return validateIgn2(cfg)
 	case ign3types.Config:
-		if reflect.DeepEqual(ign3types.Config{}, cfg) {
-			return nil
-		}
-		if report := validate3.ValidateWithContext(cfg, nil); report.IsFatal() {
-			return errors.Errorf("invalid ignition V3 config found: %v", report)
-		}
-		return nil
+		return validateIgn3(cfg)
 	default:
 		return errors.Errorf("unrecognized ignition type")
 	}
 }
 
+// Validates Ignition V2 configs
+func validateIgn2(cfg ign2types.Config) error {
```

Comment:
Just curious, did you split this out to make unit testing cleaner?

Reply:
I did this more for simplification / readability reasons.